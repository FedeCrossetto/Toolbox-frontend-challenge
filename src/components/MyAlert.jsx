import React from 'react';
import Alert from 'react-bootstrap/Alert';

const alertStyle = {
  width: '90%',
  margin: 'auto',
  marginTop: 24,
  opacity : '80%'
};

const MyAlert = ({ showAlert, setShowAlert }) => {
  return (
    showAlert && (
      <Alert
        variant="danger"
        style={alertStyle}
        onClose={() => setShowAlert(false)}
        dismissible
      >
        <p>The selected file does not contain information.</p>
      </Alert>
    )
  );
};

export default MyAlert;
