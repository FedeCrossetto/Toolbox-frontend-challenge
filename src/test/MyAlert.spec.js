import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import MyAlert from '../components/MyAlert';

describe('MyAlert component', () => {
  test('renders correctly', () => {
    const { container } = render(<MyAlert showAlert={true} setShowAlert={() => {}} />);
    expect(container).toMatchSnapshot();
  });

  test('shows alert message', () => {
    const setShowAlert = jest.fn();
    const { getByText, getByRole } = render(<MyAlert showAlert={true} setShowAlert={setShowAlert} />);

    const closeButton = getByRole('button');
    fireEvent.click(closeButton);

    expect(setShowAlert).toHaveBeenCalledWith(false);
    expect(getByText('The selected file does not contain information.')).toBeInTheDocument();
  });

  test('does not show alert message', () => {
    const { queryByText } = render(<MyAlert showAlert={false} setShowAlert={() => {}} />);
    expect(queryByText('The selected file does not contain information.')).toBeNull();
  });
});