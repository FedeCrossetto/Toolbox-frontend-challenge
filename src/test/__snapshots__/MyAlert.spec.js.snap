// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyAlert component renders correctly 1`] = `
<div>
  <div
    class="fade alert alert-danger alert-dismissible show"
    role="alert"
    style="width: 90%; margin: 24px auto auto auto;"
  >
    <button
      aria-label="Close alert"
      class="btn-close"
      type="button"
    />
    <p>
      The selected file does not contain information.
    </p>
  </div>
</div>
`;
